src/result.rs

```diff
@@ -81,6 +81,12 @@ pub struct QueryResult<'a> {
     stream: QueryStream<'a>,
 }
 
+impl<'a> Debug for QueryResult<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "QueryResult")
```

Comment:
A better option here could be to use [`fmt::DebugStruct`](https://doc.rust-lang.org/std/fmt/struct.DebugStruct.html):

```suggestion
        f.debug_struct("QueryResult").finish()
```